.github/workflows/on_push_main.yml

```diff
@@ -163,7 +170,7 @@ jobs:
     secrets: inherit
 
   build-rerun-cli-and-upload-windows:
-    needs: [checks]
+    needs: [checks, checks_cpp]
```

Comment:
This creates pipelining, adding end-to-end build time.

Why? Are these jobs using some output from `checks_cpp`?

Reply:
no, but they also don't use anything from checks do they? Thought this is so we don't do unnecssary work if stuff fails